.github/workflows/workflow-run-docker-rust-build.yaml

```diff
@@ -118,4 +199,9 @@ jobs:
             sudo systemctl stop docker
             sudo tar --posix -cf /home/runner/docker-cache.tzst -P -C /var/lib/docker --use-compress-program zstdmt .
           fi
-
+      
+      - name: Post-run Lock Cleanup 
+        if: ${{ always() && steps.lock.outcome == 'success' }}
```

Comment:
It's possible that `steps.lock` fails halfway and never writes the `outcome`. I think it's better to remove that second condition and just do `if: always()`, especially if this step succeeds always anyways with `|| true`